backends/ebpf/runtime/ebpf_runtime.c

```diff
@@ -14,89 +14,192 @@ See the License for the specific language governing permissions and
+        }
+        if (debug)
+            printf("Result of the eBPF parsing is: %d\n", result);
+    }
+    return output_pkts;
+}
+
+static void write_pkts_to_pcaps(const char *pcap_base, pcap_list_array_t *output_array) {
+    uint16_t arr_len = get_list_array_length(output_array);
+    for (uint16_t i = 0; i < arr_len; i++) {
+        char pcap_out_name[FILE_NAME_MAX];
+        if (snprintf(pcap_out_name, FILE_NAME_MAX, "%s%d%s", pcap_base, i, PCAPOUT) >=FILE_NAME_MAX) {
+            fprintf(stderr, "Filename %s%d%s too long. Truncated.\n"
+                ,pcap_base, i, PCAPIN);
+            exit(EXIT_FAILURE);
+        }
+        if (debug)
+            printf("Processing output file: %s\n", pcap_out_name);
+        pcap_list_t *out_list = get_list(output_array, i);
+        write_pkts_to_pcap(pcap_out_name, out_list);
+    }
+}
+
+static void *run_and_record_output(const char *pcap_base, pcap_list_t *pkt_list) {
+    /* Create an array of packet lists */
+    pcap_list_array_t *output_array = NULL;
+    /* Feed the packets into our "loaded" program */
+    pcap_list_t *output_pkts = feed_packets(pkt_list);
+    /* Split the output packet list by interface. This destroys the list. */
+    output_array = delete_and_split_list(output_pkts, output_array);
+    /* Write each list to a separate pcap output file */
+    write_pkts_to_pcaps(pcap_base, output_array);
+    /* Delete the array, including the data it is holding */
+    delete_array(output_array);
+}
+
+static pcap_list_t *get_packets(const char *pcap_base, uint16_t num_pcaps, pcap_list_t *merged_list) {
+    pcap_list_array_t *tmp_list_array = allocate_pkt_list_array();
+    /* Retrieve a list for each file and append it to the temporary array */
+    for (uint16_t i = 0; i < num_pcaps; i++) {
+        char pcap_in_name[FILE_NAME_MAX];
+        if (snprintf(pcap_in_name, FILE_NAME_MAX, "%s%d%s", pcap_base, i, PCAPIN) >=FILE_NAME_MAX) {
+            fprintf(stderr, "Filename %s%d%s too long. Truncated.\n"
+                ,pcap_base, i, PCAPIN);
+            exit(EXIT_FAILURE);
+        }
+
         if (debug)
-            printf("\nResult of the eBPF parsing is: %d\n", result);
+            printf("Processing input file: %s\n", pcap_in_name);
+        pcap_list_t *pkt_list = read_pkts_from_pcap(pcap_in_name, i);
+        tmp_list_array = insert_list(tmp_list_array, pkt_list, i);
+    }
+    /* Merge the array into a newly allocated list. This destroys the array. */
+    return delete_and_merge_pcap_lists(tmp_list_array, merged_list);
+}
+
+void launch_runtime(const char *pcap_name, uint16_t num_pcaps) {
+    /* Initialize the list of input packets */
+    pcap_list_t *input_list = NULL;
+    if (num_pcaps == 0)
+        return;
+
+    /* Create the basic pcap filename from the input */
+    const char *suffix = strrchr(pcap_name, DELIM);
+    if (suffix == NULL) {
+        fprintf(stderr, "Delimiter not found in %s! Exiting...\n",pcap_name);
```

Comment:
You should give a better error message: "expected a filename of the form ...".
It is not clear what delimiter you are looking for from this message.